add box for comments

hoil200 opened the pull request.

sylvia-hoang commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -44,19 +58,12 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     //response.setContentType("text/html;");
     //response.getWriter().println(show);
```
Please remove all the unused code instead of commenting them out.

sylvia-hoang commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -28,14 +28,28 @@
       shows.add("Grey\'s Anatomy");
       shows.add("House");
       shows.add("Code Black");
+  }*/
```
Remove unused code in comments.

sylvia-hoang commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -28,14 +28,28 @@
+
+      response.setContentType("text/html;");
+      response.getWriter().println(json);
   }
```
So you actually don't have to comment out the doGet() function in order to add the doPost() one! You need to keep the doGet() function since it's the one that will return the value of your `shows` variable to the frontend.

All you need to do is to remove the hardcoded values for `shows`, which you already did. Please restore your doGet() function by removing the comment slashes :)

hoil200 commented:
I removed the unused code and made the doGet function to show comments. I added the datastore to store comments and show comments. I also updated the layout of the portfolio.

sylvia-hoang commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -26,37 +33,21 @@
-    //String show = shows.get((int) (Math.random() * shows.size()));
-    //response.setContentType("text/html;");
-    //response.getWriter().println(show);
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Hey Anesia, no need to remove the doGet() function! You just have to add doPost() as a new function to the servlet. The SubtractionServlet.java file in the walkthrough contains both, as an example.

All you need to do to your doGet() function is to remove the step where you add your hardcoded messages, which you already did.

sylvia-hoang commented on portfolio/src/main/java/com/google/sps/servlets/DataServletStore.java:
```diff
@@ -0,0 +1,67 @@
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);;
```
Remove one extra semi colon.

sylvia-hoang commented on portfolio/src/main/java/com/google/sps/servlets/DataServletStore.java:
```diff
@@ -0,0 +1,67 @@
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/dataStore")
+public class DataServletStore extends HttpServlet {
```
Rename the file and class to DataStoreServlet. The convention is that you end the file name with .*Servlet

sylvia-hoang commented on portfolio/src/main/java/com/google/sps/servlets/DataServletStore.java:
```diff
@@ -0,0 +1,67 @@
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+}
```
Ok I see that you're trying to follow the example and creating 2 servlets to handle the POST and GET actions separately, and that's fine! We can keep it this way and let's try to make this work :)

sylvia-hoang commented on portfolio/src/main/webapp/index.html:
```diff
@@ -24,8 +24,25 @@ <h1 style="background-color:SlateBlue;">Anesia's Portfolio</h1>
+          <input type = "submit" />
+        </form>
+        <br/>
+        <form action = "/dataStore" method "post">
```
Your /dataStore path is associated with the DataStore only handles a GET method so there's no need to wrap the Show Comments button in a form element. loadComments() in script.js will fetch the /dataStore path and that's enough. Please remove it.

sylvia-hoang commented on portfolio/src/main/webapp/script.js:
```diff
@@ -27,25 +27,42 @@ function addRandomGreeting() {
-    fetch('/data').then(respones => response.json()).then((shows) =>{
-        console.log('test');
+    fetch('/dataStore').then(response => response.json()).then((shows) =>{
+        //console.log('test');
```
Please remove all the unused code that's commented out.

hoil200 commented:
I made the suggested changes. I not sure if I may have missed something, but I tried to correct the one's you pointed out.
